nixos/modules/system/etc/setup-etc.pl

```diff
@@ -1,5 +1,6 @@
 #! @perl@/bin/perl
 
+use feature qw(signatures);
```

Comment:
```suggestion
use experimental qw(signatures);
```

To avoid the warnings.

Also, you can use `use v5.34;` instead of `use strict`;